app/controllers/categories_controller.rb

```diff
@@ -9,4 +9,19 @@ def index
     categories = categories.where(top_level: top_level) unless top_level.nil?
     render json: CategoryPresenter.present(categories)
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def counts
+    if !admin_signed_in?
+      render status: :unauthorized
+    else
+      render status: :ok, json:
+          Category.order(:name).map { |c|
+            { name: c.name,
+              services: Service.joins(:categories).where('categories.id' => c.id).where('status' => 1).count,
+              resources: Resource.joins(:categories).where('categories.id' => c.id).where('status' => 1).count }
```

Comment:
You don't need to do the manual join here since we define the [many to many relationship](https://github.com/ShelterTechSF/askdarcel-api/blob/54b93b7fb279fd54f3471667ab932c4c3e53195a/app/models/category.rb#L2-L3) in the model. I think this can be written more succinctly like this:

```ruby
c.services.where(status: 1).count
```

Reply:
Ah, ill test that.

Reply:
Nice, that works.